### DISCUSSION TITLE
How can this software be set up for simultaneous use by multiple users?

### DISCUSSION THREAD

**Discussion by drwslacy47** [vladmandic/sdnext#3452] [General]
I'd like to run this software on a single host but provide the ability for multiple users to each use their own instance of the UI.  How can I do this?  What would happen if I launched a single instance of the UI and then different users access it at the same IP address and port the UI is running on?  Will this work?  Or is there some way to create a common server process (perhaps using the --api_only switch) and then somehow have some sort of client UI communicate with the server?

**Comment by vladmandic** [comment]
with a standard built-in ui, no, its a single-user interface.
running it as a server and performing all requests via api, absolutely, there are plenty of online services that have their custom front end and using api to sdnext in the backend.

**Reply by drwslacy47** [reply]
Thank you for the quick response!  A couple of follow-ups:

1.  Could you point me to any existing examples of a front end using the api to sdnext as the backend server?  Or stated another way, are there any out-of-the-box, open-source front ends I could lay my hands on to get up and running in this mode?  Basically, I want to build my own, locally-hosted online service that others can use.  (I like the flexibility of sdnext and the fact that it supports HF diffusers.)

2.  Would the sdnext api allow me to do everything that the sdnext single-user UI allows?  For example: model selection, adjustment of model/schedule parameters for operations such as text-to-image and image-to-image, etc.

Thank you again for taking time to respond!  I appreciate it.

**Reply by vladmandic** [reply]
re: 1 - i'm not aware of any open-sourced ones. i have access to few, but cannot share.
re: 2 - yes, api is full-featured and there is pretty much nothing that cannot be done via api.

btw, you can take a look at some examples of api usage in `cli/simple_*.py`
